torch/_dynamo/variables/misc.py

```diff
@@ -234,11 +234,63 @@ def visit(node):
 
         VariableTracker.apply(visit, (args, kwargs))
 
+        ctx = AutogradFunctionContextVariable.create(tx)
+        args = [ctx, *args]
+
         if requires_grad and torch.is_grad_enabled():
-            # TODO(jansel): handle this in training mode
-            unimplemented("autograd.Function with requires_grad")
+            # Note - this is the same check used in autograd/function.py, except inverted.
+            # If we want to support functorch transforms here, we will need to enable this.
+            if (
```

Comment:
Not sure if it's there already but we want an assert that neither `vjp` or `jvp` was re-defined on this class either.
The first one because we only look at .backward instead of doing https://github.com/pytorch/pytorch/blob/7a15e823885ddad1696e507a642fc89029a75009/torch/autograd/function.py#L273 and the second because we just ignore forward AD grads.